File: src/Java/598.RangeAdditionII.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int maxCount(int m, int n, int[][] ops) {
        if (ops.length == 0)
            return m * n;
        int minm = ops[0][0];
        int minn = ops[0][1];
        for (int[] each : ops) {
            minm = Math.min(minm, each[0]);
            minn = Math.min(minn, each[1]);
        }
        return minm * minn;
    }
}

File: src/CSharp/598.RangeAdditionII.cs


using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.RangeAdditionII
{
    public class Solution
    {
        public int MaxCount(int m, int n, int[,] ops)
        {
            int minA = m, minB = n;
            for (int i = 0; i < ops.GetLength(0); i++)
            {
                if (minA > ops[i, 0])
                    minA = ops[i, 0];
                if (minB > ops[i, 1])
                    minB = ops[i, 1];
            }
            return minA * minB;
        }
    }
}

